tensorly/metrics/losses.py

```diff
@@ -0,0 +1,97 @@
+import tensorly as tl
+import math
+
+
+def loss_operator(tensor, estimated_tensor, loss, mask=None):
+    """
+    Operator to use loss functions from [1] in order to compute loss for
+    generalized parafac decomposition.
+
+    Parameters
+    ----------
+    tensor : ndarray
+    estimated_tensor : ndarray
+    loss : {'gaussian', 'gamma', 'rayleigh', 'poisson_count', 'poisson_log', 'bernoulli_odds', 'bernoulli_log'}
+    mask : ndarray
+        array of booleans with the same shape as ``tensor`` should be 0 where
+        the values are missing and 1 everywhere else.
+
+    Returns
+    -------
+    error : ndarray
+         Size based normalized loss for each entry
+    References
+    ----------
+    .. [1] Hong, D., Kolda, T. G., & Duersch, J. A. (2020).
+           Generalized canonical polyadic tensor decomposition. SIAM Review, 62(1), 133-163.
+    """
```

Comment:
I wonder whether we are going to actually need this somewhere else? (If not, it could probably be, along with the gradient, directly in the function)